src/hssm/distribution_utils/dist.py

```diff
@@ -499,84 +174,37 @@ class HSSMRV(RandomVariable):
+        # `size` argument that is passed
+        # and the parameter vector shapes that are passed?
         @classmethod
         def rng_fn(
             cls,
             rng: np.random.Generator,
             *args,
             **kwargs,
         ) -> np.ndarray:
-            """Generate random variables from this distribution.
-
-            Parameters
-            ----------
-            rng
-                A `np.random.Generator` object for random state.
-            args
-                Unnamed arguments of parameters, in the order of `_list_params`, plus
-                the last one as size.
-            kwargs
-                Other keyword arguments passed to the ssms simulator.
-
-            Returns
-            -------
-            np.ndarray
-                An array of `(rt, response)` generated from the distribution.
-
-            Note
-            ----
-            How size is handled in this method:
-
-            We apply multiple tricks to get this method to work with ssm_simulators.
-
-            First, size could be an array with one element. We squeeze the array and
-            use that element as size.
-
-            Then, size could depend on whether the parameters passed to this method.
-            If all parameters passed are scalar, that is the easy case. We just
-            assemble all parameters into a 1D array and pass it to the `theta`
-            argument. In this case, size is number of observations.
-
-            If one of the parameters is a vector, which happens one or more parameters
-            is the target of a regression. In this case, we take the size of the
-            parameter with the largest size. If size is None, we will set size to be
-            this largest size. If size is not None, we check if size is a multiple of
-            the largest size. If not, an error is thrown. Otherwise, we assemble the
-            parameter as a matrix and pass it as the `theta` argument. The multiple then
-            becomes how many samples we draw from each trial.
-            """
-            # First figure out what the size specified here is
-            # Since the number of unnamed arguments is undetermined,
-            # we are going to use this hack.
-            size, args, kwargs = _extract_size(args, kwargs)
-
-            arg_arrays = _create_arg_arrays(cls, args)
-            p_outlier, arg_arrays = _get_p_outlier(cls, arg_arrays)
-            seed = _get_seed(rng)
-
-            is_all_args_scalar, theta, max_shape, new_data_size = (
-                _prepare_theta_and_shape(arg_arrays, size)
-            )
-            n_replicas = _calculate_n_replicas(is_all_args_scalar, size, new_data_size)
-
-            sims_out = simulator_fun_internal(
-                theta=theta,
-                random_state=seed,
-                n_replicas=n_replicas,
+            """Generate random variables from this distribution."""
+            sims_out, p_outlier = _rng_fn(
```

Comment:
Left the corresponding comment on the `ssms` side as well. 
I think `_rng_fn` shouldn't return `sims_out, p_outlier` but just `sims_out`, 
so `p_outlier` should be processed / split from the other args, before passing to `_rng_fn`. 

If you think what is done currently is necessary could you motivate / explain the choice?